tasks/extract_filepaths_for_granule/README.md

```diff
@@ -2,41 +2,163 @@
+deactivate
+cp *.py build/
+cd build
+zip -r ../extract_filepaths_for_granule.zip .
+cd ..
+```
+An example of a script for building the `extract_filepaths_for_granule` lambda package can be found in `/tasks/extract_filepaths_for_granule/bin/build_tasks.sh`.
+
+## Unit tests
+
+Unit tests for `extract_filepaths_for_granule` lambda function can be run using 
+
+```
+coverage run --source extract_filepaths_for_granule -m pytest
+```
+See the script under `tasks/extract_filepaths_for_granule/bin/run_tests.sh`
 
 ## Deployment
+
+Use the `extract_filepaths_for_granule.zip` file from build step to deploy the lambda by passing this to `filename` argument in terraform section.
+The `modules/lambdas/main.tf` shows an example of deploying this lambda through Terraform.
+
 ```
-    cd tasks\extract_filepaths_for_granule
-    rm -rf build
-    rm -rf venv
-    mkdir build
-    python3 -m venv venv
-    source venv/bin/activate
-    pip3 install -t build -r requirements.txt
-    deactivate
-    cp *.py build/
-    cd build
-    zip -r ../task.zip .
+# extract_filepaths_for_granule - Translates input for request_files lambda
+# ==============================================================================
+resource "aws_lambda_function" "extract_filepaths_for_granule" {
+  ## REQUIRED
+  function_name = "${var.prefix}_extract_filepaths_for_granule"
+  role          = module.restore_object_arn.restore_object_role_arn
+
+  ## OPTIONAL
+  description      = "Extracts bucket info and granules filepath from the CMA for ORCA request_files lambda."
+  filename         = "${path.module}/../../tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.zip"
+  handler          = "extract_filepaths_for_granule.handler"
+  memory_size      = var.orca_recovery_lambda_memory_size
+  runtime          = "python3.7"
+  source_code_hash = filebase64sha256("${path.module}/../../tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.zip")
+  tags             = local.tags
+  timeout          = var.orca_recovery_lambda_timeout
+
+  vpc_config {
+    subnet_ids         = var.lambda_subnet_ids
+    security_group_ids = [module.lambda_security_group.vpc_postgres_ingress_all_egress_id]
+  }
+}
 ```
-<a name="deployment-validation"></a>
 ### Deployment Validation
-```
+
 1.  The easiest way to test is to use the DrRecoveryWorkflowStateMachine.
     You can use the test event in tasks/extract_filepaths_for_granule/test/testevents/StepFunction.json.
     Edit the ['payload']['granules']['keys'] values as needed to be the file(s) you wish to restore.
     Edit the ['cumulus_meta']['execution_name'] to be something unique (like yyyymmdd_hhmm). Then
     copy and paste the same value to the execution name field above the input field.
 
 2.  Execute the workflow. Once it passes the extract step, you can look at the output from it.
+
+
+## Input
+The lambda handler event excepts a dictionary having a list of granules as input.
+```
```

Comment:
I would put a link and point to the input schema here along with your example.

Reply:
added